File: Python/__fundamentals/bit-manipulation/convert-to-binary/convert-to-binary.py

'''
 *  Programmer:                     Jeff C. Cheng
 *  Last modified:                  10:02PM 02-08-2024
 *  Problem:                        Convert to Binary (Zero)
 *  @param {Solution} self
 *  @param {integer} number
 *  @return {String}
'''

class Solution:
    def convert_to_binary(self, number):

#********************************************************************************************************************
#  Step 1: create an empty string
#********************************************************************************************************************

        converted = ""

#********************************************************************************************************************
#  Step 2: while the number isn't 0
#********************************************************************************************************************

        while number != 0:

#********************************************************************************************************************
#  Step 3: pre-pend a "0" or "1" based on whether the number is even or odd to the result
#********************************************************************************************************************

            if (number & 1) == 0:
                converted = "0" + converted
            else:
                converted = "1" + converted
            
            number >>= 1

#********************************************************************************************************************
#  Step 4: return the result
#********************************************************************************************************************

        return converted
    
solution = Solution

#   64  32  16  8   4   2   0
#   0   0   1   0   1   0   0
#   1   0   1   0   0   1   0

number1 = 20
number2 = 82

print(solution.convert_to_binary(solution, number1))
print(solution.convert_to_binary(solution, number2))

File: C-Class/__fundamentals/bit_manipulation/convert_to_binary/convert_to_binary.cpp

/*
 *  Programmer:                     Jeff C. Cheng
 *  Last modified:                  07:58PM 03-24-2024
 *  Problem:                        Convert to Binary (Zero)
 *  @param {Integer} number
 *  @return {Strng}
 */

#include <iostream>
#include <string>

class Solution
{
    public:
    
        std::string convert_to_binary(int num)
        {
            std::string converted = "";

            while (num != 0)
            {
                converted = (num & 1) == 0 ? ("0" + converted) : ("1" + converted);
                num >>= 1;
            }

//********************************************************************************************************************
//  Step 4: return the result
//********************************************************************************************************************

            return converted;
        }
};

int main()
{
    //  8 4 2 0
    //  1 0 0 0
    Solution solution;

    int number1 = 12;
    int number2 = 13;

    std::cout << solution.convert_to_binary(number1) << std::endl;
    std::cout << solution.convert_to_binary(number2) << std::endl;

    return 0;
}